kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/ProduceBatchAction.java

```diff
@@ -0,0 +1,434 @@
+import com.google.protobuf.ByteString;
+import io.confluent.kafkarest.Errors;
+import io.confluent.kafkarest.controllers.ProduceController;
+import io.confluent.kafkarest.controllers.RecordSerializer;
+import io.confluent.kafkarest.controllers.SchemaManager;
+import io.confluent.kafkarest.entities.EmbeddedFormat;
+import io.confluent.kafkarest.entities.ProduceResult;
+import io.confluent.kafkarest.entities.RegisteredSchema;
+import io.confluent.kafkarest.entities.v3.ProduceBatchErrorEntry;
+import io.confluent.kafkarest.entities.v3.ProduceBatchRequest;
+import io.confluent.kafkarest.entities.v3.ProduceBatchRequestEntry;
+import io.confluent.kafkarest.entities.v3.ProduceBatchResponse;
+import io.confluent.kafkarest.entities.v3.ProduceBatchResultEntry;
+import io.confluent.kafkarest.entities.v3.ProduceRequest.ProduceRequestData;
+import io.confluent.kafkarest.entities.v3.ProduceRequest.ProduceRequestHeader;
+import io.confluent.kafkarest.entities.v3.ProduceResponse.ProduceResponseData;
+import io.confluent.kafkarest.exceptions.BadRequestException;
+import io.confluent.kafkarest.exceptions.StacklessCompletionException;
+import io.confluent.kafkarest.exceptions.v3.ErrorResponse;
+import io.confluent.kafkarest.extension.ResourceAccesslistFeature.ResourceName;
+import io.confluent.kafkarest.ratelimit.DoNotRateLimit;
+import io.confluent.kafkarest.ratelimit.RateLimitExceededException;
+import io.confluent.kafkarest.resources.v3.V3ResourcesModule.ProduceResponseThreadPool;
+import io.confluent.rest.annotations.PerformanceMetric;
+import io.confluent.rest.entities.ErrorMessage;
+import io.confluent.rest.exceptions.KafkaExceptionMapper;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collector;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.validation.Valid;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.container.Suspended;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Providers;
+import org.apache.kafka.common.errors.SerializationException;
+
+@DoNotRateLimit
+@Path("/v3/clusters/{clusterId}/topics/{topicName}/records:batch")
+@ResourceName("api.v3.batch-produce.*")
+public final class ProduceBatchAction {
+
+  public static final int BATCH_ID_MINIMUM_LENGTH = 1;
+  public static final int BATCH_ID_MAXIMUM_LENGTH = 80;
+  public static final int BATCH_MAXIMUM_ENTRIES = 10;
```

Comment:
Is this number `BATCH_MAXIMUM_ENTRIES = 10` just an initial value, or drawn from some evaluation?

I think in future state, this might be a configurable parameter.